### PR TITLE
Add audio and video support to the project media area

### LINKED ISSUE #4420: Allow audio and video uploads for project media
Linked to #4115. Allow project owners/collaborators to upload example audio and video clips for use in their projects.

See also https://github.com/zooniverse/markdownz/issues/65

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://audio-video-media.pfe-preview.zooniverse.org/

Fixes #4420.

~Waiting on https://github.com/zooniverse/markdownz/pull/66 to merge.~

Describe your changes.
Allow audio and video file uploads to the project media area, and preview the files using `<audio>` or `<video>` tags.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Video upload seems to be blocked by Panoptes.
```
Handling 1 selected files
Adding 1 files
Adding media 2016-11-07 14.07.30.mp4
Creating resource for 2016-11-07 14.07.30.mp4, video/mp4
POST https://panoptes-staging.zooniverse.org/api/projects/908/attached_images 400 (Bad Request)
Got error Validation failed: Content type Content-Type must be one of image/jpeg, image/png, image/gif, image/svg+xml, audio/mpeg, audio/mp3, audio/mp4, audio/x-m4a, text/plain for 2016-11-07 14.07.30.mp4
No longer pendingFiles: 2016-11-07 14.07.30.mp4
```

**Comment by eatyourgreens** [comment]
mp3 upload works in Firefox, but fails in Chrome. I think this is a Chrome bug that we've run into before.

**Comment by eatyourgreens** [comment]
This is mostly working now. The modal popup for the audio player is annoying, so I might remove that and use it only for images.

Markdownz needs to be updated to support embedded audio before this PR will be really useful in the project builder. https://github.com/zooniverse/markdownz/issues/65

**Comment by eatyourgreens** [comment]
Updated staging to use a version of Markdownz from https://github.com/zooniverse/markdownz/pull/66 which extends the image syntax to include audio and video.

**Comment by eatyourgreens** [comment]
You can try it out here, where the third slide in the tutorial is an audio file.
https://audio-video-media.pfe-preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify

To try it out in your own projects, upload an audio file to the project media area then copy the markdown code across to a markdown field in the project somewhere.

**Comment by eatyourgreens** [comment]
Should I add a file size limit to these uploads? Some projects have run into download speed problems, where it turned they had added up to 100MB of images (in one case) to their tutorial and field guide. 

We could at least warn, if not outright block, when large files are about to be uploaded.

**Comment by srallen** [comment]
I like the idea of a warning.

**Comment by eatyourgreens** [comment]
Upgraded markdownz and restaged.

**Comment by eatyourgreens** [comment]
Apologies for the large changes to `package-lock.json`. Markdownz was installing `markdown-it` twice, so I ran `npm dedupe`.

File: app/pages/lab/media.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export default class EditMediaPage extends React.Component {
 
 EditMediaPage.defaultProps = {
   project: {},
-  validSubjectExtensions: ['.jpg', '.jpeg', '.png', '.gif', '.svg'],
+  validSubjectExtensions: ['.jpg', '.jpeg', '.png', '.gif', '.svg', '.mp3'],
```
Comment (wgranger) [inline app/pages/lab/media.jsx]: I believe this should be updated to include video extensions?

**Comment by wgranger** [comment]
What kind of video files are supported? I tried uploading a `.mov` extension and saw this warning. If there are only certain video files supported, the test on the media page ("Images can be any of: .jpg, .jpeg...") should be updated to reflect this. Also, might want to change the word "Images" since we are now allowing audio/video.

screen shot 2018-03-09 at 10 31 58 am

File: app/pages/lab/media.jsx (hunk 1 above)
Comment (eatyourgreens) [reply app/pages/lab/media.jsx]: Yep. At the moment Panoptes blocks video uploads, so I've not let project owners know that that option exists.

**Comment by eatyourgreens** [comment]
Good point about the word 'images'. That error comes back from Panoptes. I think all video uploads are blocked at the moment. I couldn't even upload a video in admin mode.

**Comment by wgranger** [comment]
Oh wait, duh, just saw the comment above. Yes, hiding video upload capabilities is a good call until the backend fix is in.

**Comment by eatyourgreens** [comment]
Rebased and re-staged to bring in the latest design changes.

**Comment by eatyourgreens** [comment]
I've opened #4458 about adding a warning for large file uploads.

I think this is ready for a final review now.

**Comment by wgranger** [comment]
What was the reason for removing the modal form for audio and video? It seems a project owner might want to preview audio contained in the media section. Removing the modal form also removed that preview, not sure if that's what was intended.

**Comment by wgranger** [comment]
LGTM. Once the conflict is resolved I think this should be good to merge.

**Comment by eatyourgreens** [comment]
👍 Rebased and re-staged to bring in the large file warning.

**Pull request merged**